src/Microsoft.NetCore.Analyzers/UnitTests/Security/DoNotUseInsecureDeserializerJavascriptSerializerWithSimpleTypeResolverTests.cs

```diff
@@ -0,0 +1,538 @@
+        }
+
+        [Fact]
+        public void DeserializeObject_DefinitelyDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace Blah
+{
+    public class Program
+    {
+        public object D(string str)
+        {
+            JavaScriptSerializer s = new JavaScriptSerializer(new SimpleTypeResolver());
+            return s.DeserializeObject(str);
+        }
+    }
+}",
+                GetCSharpResultAt(12, 20, DefinitelyRule, "object JavaScriptSerializer.DeserializeObject(string input)"));
+        }
+
+        [Fact]
+        public void DeserializeObject_AnyPath_DefinitelyDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace Blah
+{
+    public class Program
+    {
+        public object D(string str, bool flag)
+        {
+            JavaScriptSerializer s;
+            if (flag)
+                s = new JavaScriptSerializer(new SimpleTypeResolver());
+            else
+                s = new JavaScriptSerializer();
+            return s.DeserializeObject(str);
+        }
+    }
+}",
+                GetCSharpResultAt(16, 20, DefinitelyRule, "object JavaScriptSerializer.DeserializeObject(string input)"));
+        }
+
+        [Fact]
+        public void Deserialize_FromArgument_MaybeDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace Blah
+{
+    public class Program
+    {
+        public T D<T>(JavaScriptSerializer s, string str)
+        {
+            return (T) s.Deserialize(str, typeof(T));
+        }
+    }
+}",
+                GetCSharpResultAt(11, 24, MaybeRule, "object JavaScriptSerializer.Deserialize(string input, Type targetType)"));
+        }
+
+        [Fact]
+        public void Deserialize_FromField_MaybeDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace Blah
+{
+    public class Program
+    {
+        public JavaScriptSerializer Serializer;
+
+        public T D<T>(string str)
+        {
+            return (T) this.Serializer.Deserialize(str, typeof(T));
+        }
+    }
+}",
+                GetCSharpResultAt(13, 24, MaybeRule, "object JavaScriptSerializer.Deserialize(string input, Type targetType)"));
+        }
+
+        [Fact]
+        public void Deserialize_FromStaticField_MaybeDiagnostic()
+        {
+            VerifyCSharp(@"
+using System.IO;
+using System.Web.Script.Serialization;
+
+namespace Blah
+{
+    public class Program
+    {
+        public static JavaScriptSerializer Serializer;
+
+        public T D<T>(string str)
+        {
+            return (T) Program.Serializer.Deserialize(str, typeof(T));
+        }
+    }
+}",
+                GetCSharpResultAt(13, 24, MaybeRule, "object JavaScriptSerializer.Deserialize(string input, Type targetType)"));
```

Comment:
So we flag a may be diagnostic even if we have no knowledge about the serializer?

Reply:
Yep.  That's configurable, by disabling the rule.  😁